src/Shared/RoslynUtils/ParsabilityHelper.cs

```diff
@@ -101,65 +95,59 @@ private static bool IsTryParse(IMethodSymbol methodSymbol)
+        typeSymbol.AllInterfaces.Any(i => i.ConstructedFrom.EqualsByName(["System", "IParsable"]));
 
-    private static bool IsBindableViaIBindableFromHttpContext(ITypeSymbol typeSymbol, WellKnownTypes wellKnownTypes)
+    private static bool IsBindableViaIBindableFromHttpContext(ITypeSymbol typeSymbol)
     {
-        var iBindableFromHttpContextTypeSymbol = wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_IBindableFromHttpContext_T);
         var constructedTypeSymbol = typeSymbol.AllInterfaces.FirstOrDefault(
-            i => SymbolEqualityComparer.Default.Equals(i.ConstructedFrom, iBindableFromHttpContextTypeSymbol)
+            i => i.ConstructedFrom.EqualsByName(["Microsoft", "AspNetCore", "Http", "IBindableFromHttpContext"])
             );
         return constructedTypeSymbol != null &&
             SymbolEqualityComparer.Default.Equals(constructedTypeSymbol.TypeArguments[0].UnwrapTypeSymbol(unwrapNullable: true), typeSymbol);
     }
 
-    private static bool IsBindAsync(IMethodSymbol methodSymbol, ITypeSymbol typeSymbol, WellKnownTypes wellKnownTypes)
+    private static bool IsBindAsync(IMethodSymbol methodSymbol, ITypeSymbol typeSymbol)
     {
         return methodSymbol.DeclaredAccessibility == Accessibility.Public &&
             methodSymbol.IsStatic &&
             methodSymbol.Parameters.Length == 1 &&
-            SymbolEqualityComparer.Default.Equals(methodSymbol.Parameters[0].Type, wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_HttpContext)) &&
+            methodSymbol.Parameters[0].Type.EqualsByName(["Microsoft", "AspNetCore", "Http", "HttpContext"]) &&
             methodSymbol.ReturnType is INamedTypeSymbol returnType &&
-            SymbolEqualityComparer.Default.Equals(returnType.ConstructedFrom, wellKnownTypes.Get(WellKnownType.System_Threading_Tasks_ValueTask_T)) &&
+            returnType.IsGenericType &&
+            returnType.ConstructedFrom.EqualsByName(["System", "Threading", "Tasks", "ValueTask"]) &&
             SymbolEqualityComparer.Default.Equals(returnType.TypeArguments[0], typeSymbol);
     }
 
-    private static bool IsBindAsyncWithParameter(IMethodSymbol methodSymbol, ITypeSymbol typeSymbol, WellKnownTypes wellKnownTypes)
+    private static bool IsBindAsyncWithParameter(IMethodSymbol methodSymbol, ITypeSymbol typeSymbol)
     {
         return methodSymbol.DeclaredAccessibility == Accessibility.Public &&
             methodSymbol.IsStatic &&
             methodSymbol.Parameters.Length == 2 &&
-            SymbolEqualityComparer.Default.Equals(methodSymbol.Parameters[0].Type, wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_HttpContext)) &&
-            SymbolEqualityComparer.Default.Equals(methodSymbol.Parameters[1].Type, wellKnownTypes.Get(WellKnownType.System_Reflection_ParameterInfo)) &&
+            methodSymbol.Parameters[0].Type.EqualsByName(["Microsoft", "AspNetCore", "Http", "HttpContext"]) &&
```

Comment:
What happens when there are duplicate arrays? For example, `["Microsoft", "AspNetCore", "Http", "HttpContext"]` is used twice. Does it exist twice in memory?

Rather than passing arrays directly, what about having a static class with known values already defined?

Reply:
> Does it exist twice in memory?

Looks [like this is the case](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQGYACLEgYRIG8CT6zT0AWEgWQAoBKGuh/9NgAMAbQC6JCCQC8JEQCI2UAMYAnAPYBndQDMALvLgl5AQU0AHAHIBTPRXWrr8sQG4+/eoNETgMuYpUNbX1DYzMrW3tHZzd8fgBfAnigA==), although the implementation for collection expressions is changing.

I think having a static class with the known values is better for readability in either case.